### PR TITLE
[Mac] Upgrade Karabiner-DriverKit-VirtualHIDDevice to v3.1.0

### PR DESCRIPTION
**Pull request by vosaica**
Here I upgrade the Karabiner-DriverKit form v2.1.0 to v3.1.0. Some breaking changes were introduced in v3.1.0 so I have to update dext.cpp. According to few responses in #716, this upgrade works well.

### CODE REVIEW

File: doc/installation.md [hunk 1]
```diff
@@ -174,7 +174,9 @@ Simply run these commands in Windows PowerShell:
 ### macOS
 
 kmonad supports macOS 10.12 to 10.15 (Sierra, High Sierra, Mojave, and
-Catalina) and macOS 11.0 (Big Sur).
+Catalina) and macOS 11.0 (Big Sur). When using driverkit-based extension
+v2.1.0 and later, kmonad also supports macOS 16.0 (Ventura) and 17.0
+(Sonoma).
```
Comment (niklaas) [inline doc/installation.md:179]:
```suggestion
v2.1.0 and later, kmonad also supports macOS 13.0 (Ventura) and 14.0
(Sonoma).
```

I think the version numbers might be mixed up with the recent iOS releases. 🤔

**Comment by MatthiasGrandl** [comment]
I am using this PR daily and it works great!

File: doc/installation.md (hunk 1 above)
Comment (vosaica) [reply doc/installation.md]: Got it, it should be 13.0 and 14.0. I just fixed it. 👍

**Comment by slotThe** [comment]
Thanks!

**Pull request merged**